Fix dispatching of pg_constraint OID assignments.

hlinnaka opened the pull request:
Constraints are not uniquely identified by namespace and name alone.
It's possible to have multiple constraints with the same name, but attached
to different tables or domains. So include parent relation and domain OIDs
in the key.

This was uncovered by a regression test in the TINC test suite. Move the
OID consistency tests from TINC into the normal regression test suite, to
have test coverage for this.

hlinnaka commented:
This is the fix for the issue that @asubramanya raised on gpdb-dev: https://groups.google.com/a/greenplum.org/forum/#!topic/gpdb-dev/C3aI6WSxKBQ

hlinnaka commented:
@danielgustafsson, I pushed a second commit this to fix this also for pg_upgrade. However, I don't have the environment set up to conveniently test it, so I did it completely in the blind. Can you verify easily if it works?

asubramanya commented on src/test/regress/sql/oid_consistency.sql:
```diff
@@ -0,0 +1,388 @@
+group by indexrelid;
+
+-- Same for Column-Oriented AO tables
+create table indoid_co_t1(a int, b int) with(appendonly=true);
```
Comment says Column oriented, but create table is not using (orientation=column).

asubramanya commented:
+1 Thanks for fixing this.

danielgustafsson commented:
Reading the diff it seems just what I was expecting, and when testing this branch in my pg_upgrade rig and it works fine. :+1:

hlinnaka commented:
Squashed the pg_upgrade changes, and modified the test case slightly to not conflict with other tests in the regression suite. Hope it passes the PR pipeline now.
